### ISSUE TITLE
question: how would I implement a custom decorator for accessing request properties

### ISSUE THREAD

**Issue by WalterLeinert** [Romakita/ts-express-decorators#64]
Hi,

currently I'm working on adding multi client capabilities to my angular2/typescript/ts-express-decorators/knex framework. Therefore in each rest request I need the client id (which is a property of my user model) for filtering client specific data. Since I'm using passport authentication each request has a user instance in the req.user property.
My current solution would be to simply add a "@Request() request: Express.Request" parameter to each controller method. My idea to simplify this is to create some new decorator like "@RequestProperty('user') user: IUser" which injects the user parameter from the current request.

Is this a good and feasible way or do you have some other idea?

Regards,
Walter

**Comment by WalterLeinert** [comment]
Maybe I found a better solution by using @Scope/custom middleware like in https://github.com/Romakita/ts-express-decorators/issues/58 

Would this be your recommended approach?

**Comment by Romakita** [comment]
Hi @WalterLeinert,

You resolve your problem with a custom middleware : 
https://github.com/Romakita/ts-express-decorators/wiki/Middlewares

```typescript
// middlewares/customMiddleware.ts
import {IMiddleware, Middleware, EndpointInfo, Endpoint, Request} from "ts-express-decorators";

@Middleware()
export default class CustomGlobalMiddleware implements IMiddleware {
   use(@Request() request) {
       
        const request = request.body.client_id;
        request.user = {id: 'iii'};

   }
}

// in your server.ts
import {ServerLoader, ServerSettings} from "ts-express-decorators";
import Path = require("path");
const rootDir = Path.resolve(__dirname);

@ServerSettings({
   rootDir,
   acceptMimes: ['application/json'] // optional
})
export class Server extends ServerLoader {
    /**
     * This method let you configure the middleware required by your application to works.
     * @returns {Server}
     */
    $onMountingMiddlewares(): void|Promise<any> {
 
        // add your global middleware
        this.use(CustomGlobalMiddleware);
   }
}

// in controller
export class MyController {
     @Get('/')
     get(@Request request) {
           const user = request.user;
     }
}
```

Unfortunately, the custom decorator on Params are planed for v1.5. I haven't public API for create that in v1.4, sorry :/

See you,
Romain

**Comment by WalterLeinert** [comment]
Hi Romain,

thank you  for the feedback. I found a just working solution: since I'm using passport, the authenticated user is already set as property 'user' in each request.

So - as you proposed - I added the request parameter to each controller method and have access to the users clientId.

First I experimented with "@Session session" and in session.passport.user I get the userId, but not more.

Regards,
Walter
